### PR TITLE
Jenkinsjobcreation

### PR DESCRIPTION
**Pull request by ambsFlip**
This branch has the code to create a jenkins job from the the ASB. Tracked via Jira ticket https://jira.fkinternal.com/browse/PE-319  [](https://jira.fkinternal.com/browse/PE-319)

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -80,24 +81,35 @@ func getFlags() []cli.Flag {
...
+
 	flags := []cli.Flag{
 		cli.StringFlag{
-			Name:  "Pipeline Name",
-			Usage: "Name of pipeline to be created",
+			Name:  "jenkinsTemplate",
+			Usage: "Path of Jenkins template file",
+			//Destination: &pipelineName,
+		},cli.StringFlag{
+			Name:  "jenkinsURL",
```
Comment (hemantj) [inline main.go]: Show all the available jenkins instances in help

File: main.go [hunk 2]
```diff
@@ -80,24 +81,35 @@ func getFlags() []cli.Flag {
 }
 
 func getJenkinsFlags() []cli.Flag {
+
 	flags := []cli.Flag{
 		cli.StringFlag{
-			Name:  "Pipeline Name",
-			Usage: "Name of pipeline to be created",
+			Name:  "jenkinsTemplate",
```
Comment (hemantj) [inline main.go]: Use Value to take default

File: main.go [hunk 3]
```diff
@@ -80,24 +81,35 @@ func getFlags() []cli.Flag {
...
+			Usage: "Path of Jenkins template file",
+			//Destination: &pipelineName,
+		},cli.StringFlag{
+			Name:  "jenkinsURL",
+			Usage: "URL of Jenkins instance (http://host:port)",
+			//Destination: &pipelineName,
+		},cli.StringFlag{
+			Name:  "jenkinsJob",
```
Comment (hemantj) [inline main.go]: GIve standard or sample pipeline name in help

File: main.go [hunk 4]
```diff
@@ -80,24 +81,35 @@ func getFlags() []cli.Flag {
...
-			Usage: "It will exclude creation of Jenkins pipeline",
-		},
-		cli.StringSliceFlag{
-			Name:  "password",
-			Value: &cli.StringSlice{},
-			Usage: "specify debian packages that must be published Eg: lego.deb, mysqitue.deb",
-		},
+			Name:  "gitURL",
+			Usage: "Git repo URL for the new service (git@github.com:Flipkart/<git_repo_name>.git)",
+			//Destination: &gitURL,
```
Comment (hemantj) [inline main.go]: Use Detination key and reuse it in your logic, instead of reading it from context

File: main.go (hunk 2 above)
Comment (ambsFlip) [inline main.go]: mentioned in the help to leave it blank to use the default

File: main.go (hunk 1 above)
Comment (ambsFlip) [inline main.go]: Currently Added 2 instances. Going forward the code will be modified to fetch the list of jenkins instances from config and use enum to display this to use upon pressing tab

File: main.go (hunk 3 above)
Comment (ambsFlip) [inline main.go]: added a sample jenkins job name in the help file

File: main.go (hunk 4 above)
Comment (ambsFlip) [inline main.go]: done

**Pull request closed without merging**